test/runtime/jhh/colocales/colocales.py

```diff
@@ -45,10 +45,23 @@ def skipif():
     env = {k.strip():v for k,v in printchplenv.ENV_VALS.items()}
 
     # Verify Chapel configuration
+
+    # Tests requires hwloc
     if env.get('CHPL_HWLOC', 'none') == 'none':
         skipReason = "CHPL_HWLOC == none"
         return
 
+    # chpl --library is currently broken if CHPL_COMM != none
+    # and CHPL_LLVM == none
+    if (env.get('CHPL_COMM', 'none') != 'none' and
+       env.get('CHPL_LLVM', 'none') == 'none'):
+        skipReason = "CHPL_COMM != none and CHPL_LLVM == none"
+        return
+
+
+
+
+
```

Comment:
I think this would make the most sense as a suppressif. We want this to eventually work, it just doesn't right now

Reply:
I'll look into how suppressif works.

Reply:
Library testing typically needs a slightly different set up when running in multilocale versus single locale.  The tests in the interop directory are divided based on this, you might want to try a set up that is closer to that strategy to begin with rather than a suppressif

Reply:
I find our test infrastructure pretty confusing, but reading the documentation it says that `suppressif` indicates "conditions under which the test is expected to fail", whereas `skipif` indicates "conditions under which the test should not be run".  The difference is subtle, from my reading of this, `suppressif` should be used to indicate conditions under which the test **cannot** be run successfully (e.g., it will not compile correctly), whereas `skipif` indicates conditions under which the test **should not** be run (e.g., the test does not provide any additional coverage under these conditions).  However, looking through some random tests it appears that `skipif` is used more frequently than `suppressif` and it is often used it indicate when the test should not be run vs. when it cannot be run. 

I took a look at the interop tests but didn't see anything too illuminating. It's not clear what's going on there, other than the top level tests are single locale while the multilocale tests are in the multilocale directory. 

These particular tests don't need to be run in more than one configuration, as long as they are run regularly in at least one configuration. I'm not sure how to express that desire with our test infrastructure. I think probably a `skipif` with `CHPL_COMM=none`, although there might be several configurations for which that is true.

Reply:
I changed this to be a `CHPL_COMM=none` skipif.